### PR TITLE
Paramtereize manager namespaces used by functest

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
To run functest at HCO provided cluster we need to be able to change the namespace to map with clustern-network-addons-operator one and also filter pods by app=kubemacpool they don't mess with not kubemacpool pods

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/tests.go [hunk 1]
```diff
@@ -331,6 +335,14 @@ func addLabelsToNamespace(namespace string, labels map[string]string) error {
 	return err
 }
 
+func getEnv(key, fallback string) string {
```
Comment (ormergi) [inline tests/tests.go:338]: I think we should remove `fallback` parameter and initialize [this](https://github.com/k8snetworkplumbingwg/kubemacpool/pull/145/commits/74ecb40c55a63216809d28e1fa17f17714e2a64c#diff-4de094b4f2274c29e9f11102da4ca7d0R39) with the `names.MANAGER_NAMESAPCE

File: tests/tests.go [hunk 2]
```diff
@@ -28,14 +29,14 @@ import (
...
 	OtherTestNamespace     = "kubemacpool-test-alternative"
-	ManagerNamespce        = names.MANAGER_NAMESPACE
 	nadPostUrl             = "/apis/k8s.cni.cncf.io/v1/namespaces/%s/network-attachment-definitions/%s"
 	linuxBridgeConfCRD     = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s"},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"type\": \"bridge\", \"bridge\": \"br1\"}"}}`
 	podNamespaceOptInLabel = "mutatepods.kubemacpool.io"
 	vmNamespaceOptInLabel  = "mutatevirtualmachines.kubemacpool.io"
 )
 
 var (
+	managerNamespace         = getEnv("MANAGER_NAMESPACE", names.MANAGER_NAMESPACE)
```
Comment (ormergi) [inline tests/tests.go:39]: https://github.com/k8snetworkplumbingwg/kubemacpool/pull/145/commits/74ecb40c55a63216809d28e1fa17f17714e2a64c#r425070907

File: tests/tests.go (hunk 1 above)
Comment (qinqon) [reply tests/tests.go:338]: This is a generic getEnv function, we don't want to harcode it here, or do you mean to do getManagerNamespace function ?

**Comment by ormergi** [comment]
Hey I just realize we already have the manager namespace as an environment variable
we can take it from the pods like this: 
````
pods := getKubemacpoolPods
	for envVar := range pods.Items[0].Spec.Containers[0].Env {
		if envVar.Name == "POD_NAMESPCAE" {
			managerNamespce = envVar.Value
		}
	}
````
What do you think?

**Comment by qinqon** [comment]
@ormergi this code is for golang code that runs inside the pod, not at the e2e tests.

File: README.md [hunk 3]
```diff
@@ -306,3 +306,14 @@ make cluster-down
 # Run function test
 make functest
 ```
+
+In just the functest want to be deployed at pre-deployed external cluster
```
Comment (ormergi) [inline README.md:310]: How about this wording?
 "If you would like run e2e tests on an already deployed external cluster.."
or "In order to run e2e tests on external cluster.."

File: tests/tests.go [hunk 4]
```diff
@@ -28,14 +29,16 @@ import (
...
 	nadPostUrl             = "/apis/k8s.cni.cncf.io/v1/namespaces/%s/network-attachment-definitions/%s"
 	linuxBridgeConfCRD     = `{"apiVersion":"k8s.cni.cncf.io/v1","kind":"NetworkAttachmentDefinition","metadata":{"name":"%s","namespace":"%s"},"spec":{"config":"{ \"cniVersion\": \"0.3.1\", \"type\": \"bridge\", \"bridge\": \"br1\"}"}}`
 	podNamespaceOptInLabel = "mutatepods.kubemacpool.io"
 	vmNamespaceOptInLabel  = "mutatevirtualmachines.kubemacpool.io"
 )
 
 var (
+	// The MANAGER_NAMESPACE variables is passed to the test suite in case
+	// different kmp manager is at not default namespace
+	managerNamespace         = getEnv("MANAGER_NAMESPACE", names.MANAGER_NAMESPACE)
```
Comment (ormergi) [inline tests/tests.go:41]: Since this function is falling back to a default value how about we change its signature to 
`tryGetEnvVar(target, default string)`?
